apps/server/src/models/category.ts

```diff
@@ -0,0 +1,95 @@
+import mongoose from 'mongoose'
+import { defaultClasses, getModelForClass, prop, ReturnModelType } from '@typegoose/typegoose'
+
+export class Category extends defaultClasses.TimeStamps {
+    public _id: mongoose.Types.ObjectId
+
+    @prop({ required: true })
+    public title: string
+
+    @prop({ default: 'normal' })
+    public type: string
+
+    @prop({ required: true })
+    public code: CategoryCode
+
+    public toJSON(): object {
+        return {
+            _id: this._id,
+            title: this.title,
+        }
+    }
+
+    public static async initialize(this: ReturnModelType<typeof Category>) {
+        for (const category of Object.keys(fixedCategory)) {
+            const exists = await this.exists(fixedCategory[category])
+            if (!exists) {
+                await this.create(category)
+            }
+        }
+    }
+
+    public static async getCategoryById(this: ReturnModelType<typeof Category>, id: string): Promise<Category> {
+        return await this.findOne({ _id: id }).exec()
+    }
+
+    public static async getCategoryByTitle(this: ReturnModelType<typeof Category>, title: string): Promise<Category> {
+        return await this.findOne({ title: title }).exec()
+    }
+
+    public static async getCategoriesByType(this: ReturnModelType<typeof Category>, type: string): Promise<Category[]> {
+        return await this.find({ type: type }).exec()
+    }
+}
+
+export const CategoryModel = getModelForClass(Category)
+
+enum CategoryCode {
+    MYEVENT = 1,
+    HOTEVENT,
+    GAME,
+    TECHNOLOGY,
+    MOVIE_TV,
+    ANIME,
+    ARTS,
+    BUSINESS,
+    COLLECTIBLES,
+    EDUCATION,
+    CAREER,
+    FASHION_BEAUTY,
+    FOOD_DRINKS,
+    HOME,
+    GARDEN,
+    MUSIC,
+    NATURE_OUTDOORS,
+    SCIENCE,
+    SPORTS,
+    VEHICLES,
+    WELLNESS,
+    OTHERS,
+}
+
+const fixedCategory = {
+    [CategoryCode.MYEVENT]: { title: '내가 등록한 행사', number: CategoryCode.MYEVENT, type: 'special' },
```

Comment:
db 필드명을 code 로 했으니 여기도 code 로 맞춰주는게 좋을거 같아요
```suggestion
    [CategoryCode.MYEVENT]: { title: '내가 등록한 행사', code: CategoryCode.MYEVENT, type: 'special' },
```